power_grid_model_c/power_grid_model/include/power_grid_model/auxiliary/serialization/deserializer.hpp

```diff
@@ -139,9 +281,19 @@ class Deserializer {
         } catch (std::exception& e) {
             handle_error(e);
         }
+        root_key_ = {};
     }
 
   private:
+    // data members are order dependent
+    // DO NOT modify the order!
```

Comment:
There are two order dependency here:

`buffer_from_json_` needs to be before `data_` and `size_`. Since we read the `msgpack` buffer twice, the buffer has to live as long as the deserializer lives. If you get a input from `json`, we need to locally store the converted `msgpack` bytes. Only after that, we can save the `data_` and `size_`.

Another dependency is: `dataset_handler_`. This has to be at the end of the members. We are calling many member functions to pre-parse the data to get the information to construct `dataset_handler_`:

```c++
dataset_handler_{pre_parse()}
```

During `pre_parse` other member data variables are used to store meta data and error diagnostic information. Other member data has to be initialized before `pre_parse` is called when the contrcutor tries to initialize `dataset_handler_`.

Reply:
member variables should all have a lifetime that is as long as the object right? (or at least until the end of the user-defined body, if any, of the destructor) After the deserializer object is destroyed, we shouldn't need to touch any of the variables, or am i understanding this incorrectly?